File: DriverTemplates/LocalServer Template/ReferenceCountedObject.cs
using System;
using System.Runtime.InteropServices;

namespace TEMPLATENAMESPACE
{
	[ComVisible(false)]
	public class ReferenceCountedObjectBase
	{
		public ReferenceCountedObjectBase()
		{
			// We increment the global count of objects.
			TEMPLATEDEVICENAME.CountObject();
		}

		~ReferenceCountedObjectBase()
		{
			// We decrement the global count of objects.
			TEMPLATEDEVICENAME.UncountObject();
			// We then immediately test to see if we the conditions
			// are right to attempt to terminate this server application.
			TEMPLATEDEVICENAME.ExitIf();
		}
	}
}

File: tags/Release_5.5.1_RTW/DriverTemplates/LocalServer Composite Driver VB/ReferenceCountedObject.vb

Imports System
Imports System.Runtime.InteropServices

Namespace TEMPLATENAMESPACE
	<ComVisible(False)> _
	Public Class ReferenceCountedObjectBase
		Public Sub New()
			' We increment the global count of objects.
			TEMPLATEDEVICENAME.CountObject()
		End Sub
		Protected Overrides Sub Finalize()
			Try

				' We decrement the global count of objects.
				TEMPLATEDEVICENAME.UncountObject()
				' We then immediately test to see if we the conditions
				' are right to attempt to terminate this server application.
				TEMPLATEDEVICENAME.ExitIf()
			Finally
				MyBase.Finalize()
			End Try
		End Sub
	End Class
End Namespace

'=======================================================
'Service provided by Telerik (www.telerik.com)
'Conversion powered by NRefactory.
'Built and maintained by Todd Anglin and Telerik
'=======================================================
